lua/nvim_lsp/rnix.lua

```diff
@@ -0,0 +1,84 @@
+local configs = require 'nvim_lsp/configs'
+local util = require 'nvim_lsp/util'
+
+local name = "rnix"
+
+local function make_installer()
+  local P = util.path.join
+  local install_dir = P{util.base_install_dir, name}
+
+  local bin = P{install_dir, "bin", "rnix-lsp"}
+  local cmd = {bin}
+
+  local X = {}
+  function X.install()
+    local install_info = X.info()
+    if install_info.is_installed then
+      print(name, "is already installed")
+      return
+    end
+    if not (util.has_bins("curl")) then
+      error('Need "curl" to install this.')
+      return
+    end
+    if not (util.has_bins("cargo")) then
+      error('Need "cargo" to install this.')
+      return
+    end
+
+    local install_cmd = "cargo install rnix-lsp --root=" .. install_info.install_dir .. " rnix-lsp"
+
+    vim.fn.system(install_cmd)
+  end
+  function X.info()
+    return {
+      is_installed = util.path.exists(bin);
+      install_dir = install_dir;
+      cmd = cmd;
+    }
+  end
+  function X.configure(config)
+    local install_info = X.info()
+    if install_info.is_installed then
+      config.cmd = cmd
+    end
+  end
+  return X
+end
+
+local installer = make_installer()
+
+configs[name] = {
+
+  default_config = {
+    filetypes = {"nix"};
+    root_dir = function(fname)
+      return util.find_git_ancestor(fname) or vim.loop.os_homedir()
+    end;
+    settings = {
+    };
+    on_new_config = function(config)
+      installer.configure(config)
+    end;
+    init_options = {
+    };
+  };
+  -- on_new_config = function(new_config) end;
+  -- on_attach = function(client, bufnr) end;
+  docs = {
+    description = [[
+https://github.com/nix-community/rnix-lsp
+
+A language server for Nix providing basic completion and formatting via nixpkgs-fmt.
```

Comment:
Would you add a short description how to install manually?

Reply:
Added, including a note for using with nix.